tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/op/TypedGradientAdapter.java

```diff
@@ -17,77 +17,41 @@
-//import org.tensorflow.internal.c_api.NativeStatus;
-//import org.tensorflow.internal.c_api.TF_Scope;
-//
-///** A native adapter for {@link CustomGradient}. */
-//final class TypedGradientAdapter<T extends RawOpInputs<?>> extends BaseGradientAdapter {
-//
-//  private final CustomGradient<T> gradient;
-//  private final Class<T> opInputClass;
-//  private final Constructor<T> ctor;
-//
-//  TypedGradientAdapter(CustomGradient<T> gradient, Class<T> opInputClass) {
-//    super();
-//    this.gradient = gradient;
-//    this.opInputClass = opInputClass;
-//    //noinspection unchecked
-//    this.ctor = (Constructor<T>) this.opInputClass.getDeclaredConstructors()[0];
-//  }
-//
-//  @Override
-//  public NativeStatus call(
-//      TF_Scope scope,
-//      NativeOperation op,
-//      NativeOutputVector grad_inputs,
-//      NativeOutputVector grad_outputs) {
-//    try (PointerScope pointerScope = new PointerScope()) {
-//      Graph g = Graph.findGraphForPointer(scope.graph());
-//      if (g == null) {
-//        throw new IllegalStateException("No graph found for native gradient scope.");
-//      }
-//
-//      T rawOp = ctor.newInstance(BaseGradientAdapter.getGraphOp(g, op.node()));
-//
-//      Scope nativeScope =
-//          new GradientScope(scope, g, null).withSubScope(rawOp.getOutputs().op().name());
-//
-//      Ops tf = new Ops(nativeScope);
-//
-//      List<Output<?>> gradInputs = BaseGradientAdapter.fromNativeOutputs(g, grad_inputs);
-//
-//      // The graph locks are not re-entrant, so attempting to add an op to a graph that has been
-//      // locked by the gradient builder will fail without this.
-//      BaseGradientAdapter.useDangerousLockedBuilders(g, true);
-//      List<Operand<?>> gradOutputs = gradient.call(tf, rawOp, gradInputs);
-//      BaseGradientAdapter.useDangerousLockedBuilders(g, false);
-//
-//      BaseGradientAdapter.putToNativeOutputs(gradOutputs, grad_outputs);
-//
-//    } catch (InvocationTargetException | InstantiationException | IllegalAccessException e) {
-//      throw new RuntimeException("Could not instantiate Op class " + opInputClass, e);
-//    }
-//    return org.tensorflow.internal.c_api.global.tensorflow.OkStatus();
-//  }
-//}
-// ---------------------------------------------------------
-// END OF CUSTOM GRADIENT
-// ---------------------------------------------------------
\ No newline at end of file
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.util.List;
+
+/** A native adapter for {@link CustomGradient}. */
+final class TypedGradientAdapter<T extends RawOpInputs<?>> extends AbstractGradientAdapter {
+
+  private final CustomGradient<T> gradient;
+  private final Class<T> opInputClass;
+  private final Constructor<T> ctor;
+
+  TypedGradientAdapter(CustomGradient<T> gradient, Class<T> opInputClass) {
+    super();
+    this.gradient = gradient;
+    this.opInputClass = opInputClass;
+    //noinspection unchecked
```

Comment:
Shouldn't this be `@SuppressWarnings("unchecked")`?

Reply:
That was there before...  I'll check